pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -18,70 +18,191 @@ package k8sorchestrator
+}
+
+// initFSS performs all the operations required to initialize the Feature states map and keep a watch on it
+func initFSS(ctx context.Context, k8sClient clientset.Interface, controllerClusterFlavor cnstypes.CnsClusterFlavor, params interface{}) error {
+	log := logger.GetLogger(ctx)
+	var (
+		fssConfigMap *v1.ConfigMap
+		err          error
+	)
+
+	// Store configmap info in global variables to access later
+	if controllerClusterFlavor == cnstypes.CnsClusterFlavorWorkload {
+		k8sOrchestratorInstance.supervisorFSS.featureStates = make(map[string]string)
+		// Validate init params
+		svInitParams, ok := params.(K8sSupervisorInitParams)
+		if !ok {
+			return fmt.Errorf("expected orchestrator params of type K8sSupervisorInitParams, got %T instead", params)
+		}
+		k8sOrchestratorInstance.supervisorFSS.configMapName = svInitParams.SupervisorFeatureStatesConfigInfo.Name
+		k8sOrchestratorInstance.supervisorFSS.configMapNamespace = svInitParams.SupervisorFeatureStatesConfigInfo.Namespace
+	}
+	if controllerClusterFlavor == cnstypes.CnsClusterFlavorVanilla {
+		k8sOrchestratorInstance.internalFSS.featureStates = make(map[string]string)
+		// Validate init params
+		vanillaInitParams, ok := params.(K8sVanillaInitParams)
+		if !ok {
+			return fmt.Errorf("expected orchestrator params of type K8sVanillaInitParams, got %T instead", params)
+		}
+		k8sOrchestratorInstance.internalFSS.configMapName = vanillaInitParams.InternalFeatureStatesConfigInfo.Name
+		k8sOrchestratorInstance.internalFSS.configMapNamespace = vanillaInitParams.InternalFeatureStatesConfigInfo.Namespace
+	}
+	if controllerClusterFlavor == cnstypes.CnsClusterFlavorGuest {
+		k8sOrchestratorInstance.supervisorFSS.featureStates = make(map[string]string)
+		k8sOrchestratorInstance.internalFSS.featureStates = make(map[string]string)
+		// Validate init params
+		guestInitParams, ok := params.(K8sGuestInitParams)
+		if !ok {
+			return fmt.Errorf("expected orchestrator params of type K8sGuestInitParams, got %T instead", params)
+		}
+		k8sOrchestratorInstance.internalFSS.configMapName = guestInitParams.InternalFeatureStatesConfigInfo.Name
+		k8sOrchestratorInstance.internalFSS.configMapNamespace = guestInitParams.InternalFeatureStatesConfigInfo.Namespace
+		k8sOrchestratorInstance.supervisorFSS.configMapName = guestInitParams.SupervisorFeatureStatesConfigInfo.Name
+		k8sOrchestratorInstance.supervisorFSS.configMapNamespace = guestInitParams.SupervisorFeatureStatesConfigInfo.Namespace
+	}
+
+	// Initialize supervisor FSS map values
+	if controllerClusterFlavor == cnstypes.CnsClusterFlavorGuest || controllerClusterFlavor == cnstypes.CnsClusterFlavorWorkload {
+		if k8sOrchestratorInstance.supervisorFSS.configMapName != "" && k8sOrchestratorInstance.supervisorFSS.configMapNamespace != "" {
+			// Retrieve configmap
+			fssConfigMap, err = k8sClient.CoreV1().ConfigMaps(k8sOrchestratorInstance.supervisorFSS.configMapNamespace).Get(
```

Comment:
Line 155 to 165 can be a function?

Reply:
I tried doing that, it did not help much. I would be passing a lot of arguments and the code isn't much of a repetition, just logs.